smtp/smtp.go

```diff
@@ -38,18 +38,50 @@ func formatHeader(h mail.Header) string {
 }
 
 type session struct {
-	c           *protonmail.Client
-	u           *protonmail.User
-	privateKeys openpgp.EntityList
-	addrs       []*protonmail.Address
+	c            *protonmail.Client
+	u            *protonmail.User
+	privateKeys  openpgp.EntityList
+	addrs        []*protonmail.Address
+	allReceivers []*mail.Address
 }
 
 func (s *session) Mail(from string, options smtp.MailOptions) error {
 	return nil
 }
 
-func (s *session) Rcpt(to string) error {
-	return nil
+func (s *session) Rcpt(to string) (err error) {
+	log.Printf("Sending mail for %s", to)
+	if to == "" {
+		return
+	}
+
+	// Seems like github.com/emersion/go-smtp/conn.go:487 removes marks on message
```

Comment:
`mail.ParseAddress` should be able to parse the string here, no need to do it manually.

Reply:
Yes, it works, but needs changes on `go-smtp`.
Consider this playground: https://play.golang.org/p/LegwsnQbpos

Reply:
Hm, go-smtp isn't supposed to return the `TO:` part…

Reply:
This is supposed to work: https://play.golang.org/p/qMBIcNqhvox

If it doesn't, this is a go-smtp bug.